### ISSUE TITLE
In a terminal the modeline is blank on startup

### ISSUE THREAD

**Issue by wrachwal** [milkypostman/powerline#71]
Emacs 24.4 compiled from source, checked in gnome teminal and xterm. I have to e.g. M-x C-g to get the modeline displayed.

$ emacs -nw -Q -l check-powerline.el

```elisp
(require 'package)
(setq package-user-dir "~/tmp/emacs/elpa")
(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
(package-initialize)
(dolist (pkg '(powerline))
  (when (not (package-installed-p pkg))
    (package-install pkg)))
(require 'powerline)
(powerline-default-theme)
```

**Comment by supernintendo** [comment]
Do you see anything in your messages buffer? I have the same issue and get:

`Error during redisplay: (eval (let* ((active (powerline-selected-window-active)) (mode-line (if active (quote mode-line) (quote mode-line-inactive))) (face1 (if active (quote powerline-active1) (quote powerline-inactive1))) (face2 (if active (quote powerline-active2) (quote powerline-inactive2))) (separator-left (intern (format "powerline-%s-%s" powerline-default-separator (car powerline-default-separator-dir)))) (separator-right (intern (format "powerline-%s-%s" powerline-default-separator (cdr powerline-default-separator-dir)))) (lhs (list (powerline-raw "%*" nil (quote l)) (powerline-buffer-size nil (quote l)) (powerline-buffer-id nil (quote l)) (powerline-raw " ") (funcall separator-left mode-line face1) (powerline-narrow face1 (quote l)) (powerline-vc face1))) (rhs (list (powerline-raw global-mode-string face1 (quote r)) (powerline-raw "%4l" face1 (quote r)) (powerline-raw ":" face1) (powerline-raw "%3c" face1 (quote r)) (funcall separator-right face1 mode-line) (powerline-raw " ") (powerline-raw "%6p" nil (quote r)) (powerline-hud face2 face1))) (center (list (powerline-raw " " face1) (funcall separator-left face1 face2) (when (boundp (quote erc-modified-channels-object)) (powerline-raw erc-modified-channels-object face2 (quote l))) (powerline-major-mode face2 (quote l)) (powerline-process face2) (powerline-raw " :" face2) (powerline-minor-modes face2 (quote l)) (powerline-raw " " face2) (funcall separator-right face2 face1)))) (concat (powerline-render lhs) (powerline-fill-center face1 (/ (powerline-width center) 2.0)) (powerline-render center) (powerline-fill face1 (powerline-width rhs)) (powerline-render rhs)))) signaled (void-variable powerline-selected-window)`

**Comment by Archenoth** [comment]
I get this issue too... Though it would only happen the first time it tried to load, and doing something that would update the modeline would fix it for the rest of the Emacs session.

With the recent update, now it just stays broken and I get the following in my messages buffer:

``
Error during redisplay: (eval (let* ((active (powerline-selected-window-active)) (mode-line (if active (quote mode-line) (quote mode-line-inactive))) (face1 (if active (quote powerline-active1) (quote powerline-inactive1))) (face2 (if active (quote powerline-active2) (quote powerline-inactive2))) (separator-left (intern (format "powerline-%s-%s" (powerline-current-separator) (car powerline-default-separator-dir)))) (separator-right (intern (format "powerline-%s-%s" (powerline-current-separator) (cdr powerline-default-separator-dir)))) (lhs (list (powerline-raw "%*" nil (quote l)) (powerline-buffer-size nil (quote l)) (powerline-buffer-id nil (quote l)) (powerline-raw " ") (funcall separator-left mode-line face1) (powerline-narrow face1 (quote l)) (powerline-vc face1))) (rhs (list (powerline-raw global-mode-string face1 (quote r)) (powerline-raw "%4l" face1 (quote r)) (powerline-raw ":" face1) (powerline-raw "%3c" face1 (quote r)) (funcall separator-right face1 mode-line) (powerline-raw " ") (powerline-raw "%6p" nil (quote r)) (powerline-hud face2 face1))) (center (list (powerline-raw " " face1) (funcall separator-left face1 face2) (when (boundp (quote erc-modified-channels-object)) (powerline-raw erc-modified-channels-object face2 (quote l))) (powerline-major-mode face2 (quote l)) (powerline-process face2) (powerline-raw " :" face2) (powerline-minor-modes face2 (quote l)) (powerline-raw " " face2) (funcall separator-right face2 face1)))) (concat (powerline-render lhs) (powerline-fill-center face1 (/ (powerline-width center) 2.0)) (powerline-render center) (powerline-fill face1 (powerline-width rhs)) (powerline-render rhs)))) signaled (void-variable left) [8 times]
``

**Comment by milkypostman** [comment]
Should be fixed in d9fc947bb1f245333832f762a054b10496adecc2
